src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/Rules/CannotSealTypeTests.cs

```diff
@@ -0,0 +1,381 @@
+                {
+                     new CompatDifference(DiagnosticIds.CannotSealType, string.Empty, DifferenceType.Changed, "T:CompatTests.UnsealedClass")
+                };
+
+                Assert.Equal(expected, differences, CompatDifferenceComparer.Default);
+            }
+
+        }
+
+        [Fact]
+        public void MultipleRightsAreReportedCorrectly()
+        {
+            string leftSyntax = @"
+namespace CompatTests
+{
+    public class UnsealedClass
+    {
+    }
+}
+";
+
+            string[] rightSyntaxes = new[]
+            {
+                @"
+namespace CompatTests
+{
+    public class UnsealedClass
+    {
+    }
+}",
+                @"
+namespace CompatTests
+{
+    public class UnsealedClass
+    {
+        protected UnsealedClass() { }
+    }
+}",
+                @"
+namespace CompatTests
+{
+    public sealed class UnsealedClass
+    {
+    }
+}",
+                @"
+namespace CompatTests
+{
+    public class UnsealedClass
+    {
+        private UnsealedClass() { }
+    }
+}",
+            };
+
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax);
+            MetadataInformation leftMetadata = new("left", "net6.0", "ref/a.dll");
+            ElementContainer<IAssemblySymbol> leftContainer = new(left, leftMetadata);
+
+            IList<ElementContainer<IAssemblySymbol>> right = SymbolFactory.GetElementContainersFromSyntaxes(rightSyntaxes);
+
+            ApiComparer differ = new();
+            IEnumerable<(MetadataInformation left, MetadataInformation right, IEnumerable<CompatDifference> differences)> result =
+                differ.GetDifferences(leftContainer, right);
+
+            CompatDifference[][] expectedDiffs =
+            {
+                Array.Empty<CompatDifference>(),
+                Array.Empty<CompatDifference>(),
+                new[]
+                {
+                    new CompatDifference(DiagnosticIds.CannotSealType, string.Empty, DifferenceType.Changed, "T:CompatTests.UnsealedClass"),
+                },
+                new[]
+                {
+                    new CompatDifference(DiagnosticIds.CannotSealType, string.Empty, DifferenceType.Changed, "T:CompatTests.UnsealedClass"),
+                },
+            };
+
+            AssertExtensions.MultiRightResult(leftMetadata, expectedDiffs, result);
+        }
+
+        [Fact]
+        public void MultipleRightsNoDifferences()
+        {
+            string leftSyntax = @"
+namespace CompatTests
+{
+    public class UnsealedClass
+    {
+    }
+}
+";
+
+            string[] rightSyntaxes = new[]
+            {
+                @"
+namespace CompatTests
+{
+    public class UnsealedClass
+    {
+    }
+}",
+                @"
+namespace CompatTests
+{
+    public class UnsealedClass
+    {
+        protected UnsealedClass() { }
+    }
+}",
+                @"
+namespace CompatTests
+{
+    public class UnsealedClass
+    {
+        internal UnsealedClass() { }
```

Comment:
Shouldn't this be a breaking change, since it removed the public default constructor?

Reply:
Yes I thought the same. Looked at it, it is because it is finding a matching .cctor on the base type because the name, return type and parameters are the same. I need to condition that to not do it for constructors.